src/importlinter/domain/fields.py

```diff
@@ -70,6 +70,25 @@ def parse(self, raw_data: Union[str, List]) -> List[Any]:
         return clean_list
 
 
+class SetField(ListField):
+    """
+    A field for multiple, unique values of any type.
+
+    Arguments:
+        - subfield: An instance of a single-value Field. Each item in the list will be the return
+                    value of this subfield.
+    Usage:
+
+        field = SetField(subfield=AnotherField())
+
+    """
+
+    def parse(  # type: ignore[override]  # noqa: F821
```

Comment:
@seddonym I found this implementation the clearest one, but it's not valid according to `mypy`, so I can update it to use composition instead of inheritance - we can create `ListField` in `__init__` and use it in `parse()` to avoid subclassing `ListField`